timApp/static/scripts/tim/gamification/badge/badge.leaderboard.ts

```diff
@@ -0,0 +1,185 @@
+import {Component, Input, NgModule} from "@angular/core";
+import {OnInit} from "@angular/core";
+import {CommonModule} from "@angular/common";
+import {FormsModule} from "@angular/forms";
+import {HttpClient} from "@angular/common/http";
+import {BadgeService} from "tim/gamification/badge/badge.service";
+import {firstValueFrom} from "rxjs";
+import {Subscription} from "rxjs";
+import {GroupService} from "tim/plugin/group-dashboard/group.service";
+import {TimUtilityModule} from "tim/ui/tim-utility.module";
+
+@Component({
+    selector: "tim-badge-leaderboard",
+    template: `
+        <link href="https://fonts.googleapis.com/css2?family=Material+Symbols+Outlined" rel="stylesheet" />        
+        <div class="viewer-container">             
+            <h2>{{badgegroupContext}} top5 Leaderboard</h2>
```

Comment:
Voi olla että pienessä/isossa coding campissä 5 parasta on liian paljon tai liian vähän. Vois jatkokehitykseen kirjata että pitäis voida säätää kuinka monta parasta näytetään